ethyca/fides
# Audit Command for Resource Compliance

Closes #372

### Code Changes

* [x] Add a new option to `--include-audit` of server resources when running `evaluate`
* [x] Add validation of a systems attributes that support the data map
* [x] Add validation for Organization attributes
* [x] ~Add validation of a dataset attribute~ not required at this time
* [ ] Remove the explicit `audit` command left from earlier

### Steps to Confirm

* [x] `fidesctl evaluate demo_resources/ -a` successfully audits and returns valid and clear(er) messaging

### Pre-Merge Checklist

* Documentation Updated:
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)

### Description Of Changes

A new option to audit for compliance seemed to make sense here as a starting point. Long-term, the vision is to extend policies to handle multiple types of validation (ci checks, governance, etc.). For this reason, combining with `evaluate` instead of creating another command.

Steps to validate:
1. start up the cli with `nox -s cli`
1. `fidesctl db reset -y`
1. `fidesctl evaluate demo_resources/ -a`
1. (should return an error when auditing systems)
1. set both data_uses in `demo_system.yml` to `third_party_sharing.personalized_advertising.direct_marketing`
1. set both data_subjects in `demo_system.yml` to `potential_customer`
1. `fidesctl evaluate demo_resources/ -a` (should now succeed)
